### PR TITLE
support conditional text2image generation in worker

### PR DESCRIPTION
**Pull request by noskill**

### CODE REVIEW

File: multigen/pipes.py [hunk 1]
```diff
@@ -325,7 +325,8 @@ class Cond2ImPipe(BasePipe):
         "soft-sobel": 0.3,
         "soft": 0.95, #0.5
         "depth": 0.5,
-        "inpaint": 1.0
+        "inpaint": 1.0,
+        "qr": 1.5
```
Comment (Necr0x0Der) [inline multigen/pipes.py:329]: What is qr?

Comment (noskill) [reply multigen/pipes.py:329]: yes, it this model https://huggingface.co/DionTimmer/controlnet_qrcode

File: multigen/worker.py [hunk 2]
```diff
@@ -32,14 +36,16 @@ def _update(sess, job, gs):
...
                 pipe_name = sess.get('pipe', 'Prompt2ImPipe')
-                model_id = str(self.cwd/self.config["model_dir"]/self.models['base'][sess["model"]])
-                pipe = self.get_pipeline(pipe_name, model_id)
+                model_id = str(self.cwd/self.config["model_dir"]/self.models['base'][sess["model"]]['id'])
+                pipe = self.get_pipeline(pipe_name, model_id, cnet=data.get('cnet', None))
 
                 # TODO: the list of provided images can depend on the pipeline (e.g., it will be different for Cond)
                 images = data['images']
-                if images:
+                if len(images) == 2:
```
Comment (noskill) [inline multigen/worker.py:44]: condition should be pipeline id, not number of images

Comment (Necr0x0Der) [reply multigen/worker.py:44]: I'm not sure to whom you address this comment since it is your code?

File: multigen/worker.py [hunk 3]
```diff
@@ -7,14 +7,18 @@
...
         pipe_class = self.get_pipe_class(pipe_name)
-        pipeline = self._loader.load_pipeline(pipe_class._class, model_id)
-
-        pipe = pipe_class(model_id, pipe=pipeline)
+        if cnet is None:
+            pipeline = self._loader.load_pipeline(pipe_class._class, model_id)
+            pipe = pipe_class(model_id, pipe=pipeline)
+        else:
+            pipeline = self._loader.get_pipeline(model_id)
+            if pipeline is None or 'controlnet' not in pipeline.pipe.components :
```
Comment (noskill) [inline multigen/worker.py]: additional check is needed: controlnet in existing pipeline might be wrong one

**Review by Necr0x0Der — APPROVED** [review]

**Pull request merged**